internal/guest/runtime/hcsv2/container.go

```diff
@@ -176,3 +180,12 @@ func (c *Container) GetStats(ctx context.Context) (*v1.Metrics, error) {
 func (c *Container) modifyContainerConstraints(ctx context.Context, rt prot.ModifyRequestType, cc *prot.ContainerConstraintsV2) (err error) {
 	return c.Update(ctx, cc.Linux)
 }
+
+// adjustExecSpec adjusts an OCI runtime specs process field provided for an execed process to contain some of the settings set on the containers spec.
+// Some of the OCI spec settings are configured in the guest as there's not enough information available on the host to set them, so the spec set for the
+// exec may be missing some things that were present in the final version of the containers spec. This includes things such as the containers user configured
+// for the init process as we can't verify that the user exists in the container image on the host. A consequence of the user example is any exec will run as root
+// instead of the user configured for the container which can be problematic.
+func adjustExecSpec(containerSpec *oci.Process, execSpec *oci.Process) {
+	execSpec.User = containerSpec.User
```

Comment:
With the process dump work, we'd carry over the rlimits on the container here for example. https://github.com/microsoft/hcsshim/pull/1062/files#diff-6cfb31c890bf0639911770222949ee54db473a0945fea0ff1a6c7ef1cf7f9db8R69-R71

Reply:
Is there a possibility that one would want to run a process inside a container with a different user than the one which was used in the container spec? To handle that case maybe we should check if `execSpec.User` is set already and change it here only if it is not set?

Reply:
That's a possibility, nothing stopping a client of containerd to make this functionality it's just not exposed through cri/ctr which is what I'm basing this off of. I'll look into adding and respond shortly, thanks!

Reply:
It'll be hard to check if the user is set, as it's not a pointer and the default is everything 0 and empty string (and uid 0 and gid 0 translate to root lol). I can check if the exec specs fields match that of the containers and if not then prefer the execs though.

Reply:
Doing now

Reply:
Ahh, nevermind! Ctr does allow you to set a user for an exec only cri doesn't, so this would be good to add. I guess I'll treat anything that's not uid 0/gid 0 as the user attempted to set something and compare if the container and execs match, and if not then don't replace.

Reply:
That's actually kind of annoying that the cri api doesn't take a user string in the exec structs.. Maybe we should look at adding that

Reply:
Pushed, let me know if this addresses your comment.